docs/src/documentation/03-components/04-overlay/tooltip/03-accessibility.mdx

```diff
@@ -0,0 +1,84 @@
+---
+title: Accessibility
+redirect_from:
+  - /components/tooltip/accessibility/
+---
+
+# Accessibility
+
+## Tooltip
+
+The Tooltip component has been designed with accessibility in mind.
+
+It can be used with keyboard navigation, and it includes the following properties that allow to improve the experience for users of assistive technologies:
+
+| Name            | Type     | Description                                                                               |
+| :-------------- | :------- | :---------------------------------------------------------------------------------------- |
+| aria-label      | `string` | Adds `aria-label` attribute to the rendered tooltip element. Announced by screen readers. |
+| aria-labelledby | `string` | Id(s) of elements that announce the component to screen readers.                          |
+
+Mentioned props are optional, but highly recommended to use to ensure the best accessibility experience for all users.
+
+The `aria-label` prop can be used to provide a label for the tooltip that is announced by screen readers.
+
+The `aria-labelledby` prop can reference multiple ids, separated by a space. Value of the attribute is not announced by screen reader, content of connected
+element does.
+
+The conjugation of these properties allows to provide a content of tooltip component to users of assistive technologies.
+
+For example, the following code snippet shows how to use these properties:
+
+```jsx
+<Tooltip
+  content={
+    <div>
+      <p>Lorem ipsum dolor sit amet</p>
+    </div>
+  }
+  aria-label="Tooltip label"
+  aria-labelledby="tooltip-labelledby"
+>
+  <Text id="tooltip-labelledby">Hello world.</Text>
+</Tooltip>
+```
+
+Note that attribute `aria-labelledby` is connected with `id` attribute of the component that is the base for the Tooltip. Without that connection, the `aria-labelledby` attribute is ignored and not read by screen reader.
+
+The content of attribute `aria-labelledby` is prioritized before the content of `aria-label`. In case `aria-labelledby` is not filled in,
+then the content of `aria-label` is read by screen reader.
+
+Be sure to include all relevant information on all of the properties that are announced by screen readers.
+
+### Examples
+
+1. Prop `id` in `<Text>` component is not filled in, content of `aria-label` is read by screen reader. Then the content inside the tooltip is read by screen reader. Content of `aria-labelledby` is ignored.
```

Comment:
This is misleading. The content of `aria-label` is read because there is no element with the `id` passed on `aria-labelledby`. It is not because `Text` has no `id`.

It is also important to note that the paragraph inside the content is also announced (although it shouldn't…). So this is a wrong implementation that we currently have and that we will need to fix. But until that happens I feel we should be clear about it, and adjust the documentation when the time comes.

Reply:
Can you elaborate on the problem here, please? By "paragraph inside the content" you mean specifically the "Lorem ipsum dolor sit amet" text from the example? Or that the word "paragraph" as an element/role is being announced? Or what is the problem we need to fix?

Reply:
The `"Lorem ipsum dolor sit amet"` is also announced because the tooltip is focused when it's open (it shouldn't, but that's an issue for another moment)

Reply:
Isn't that the desired behavior to announce the tooltip content (at least I was under the impression it is, from reading the W3 materials)? Can you share some resources on this, please?

Reply:
The content of the tooltip **must** be announced, but not the way we are doing it, I suppose. Now with the migration to floating-ui things are a bit better, but in the past they were announced because we forced focus on the element, which should not be the case…

I am just saying that: if we have the content already being announced naturally (even if the implementation isn't perfect), what is the point of the aria-label? What should one put on the `aria-label` to be announced that is not announced by the content itself?

Reply:
> I get the point that aria-label is not needed (?) in this case when the tooltip content itself is read anyway.

Now I am confused by _you_ 😅

From my understanding, the `aria-label`/`aria-labelledby` is not needed when reference element contains understandable text:

![Snímek obrazovky 2024-11-20 v 15 39 11](https://github.com/user-attachments/assets/5c6a5e92-98dc-469f-99f6-04c294bb5a28)

but is needed when we use the icon:

![Snímek obrazovky 2024-11-20 v 15 36 40](https://github.com/user-attachments/assets/b89a1d6c-187c-4045-8750-aa8bf0d8bbf2)

But maybe if the icon has accessible text already, `aria-label` on the Tooltip reference element is not needed at all?

Reply:
yes, that's what I mean, sorry for the confusion. :D

Reply:
Back to the first comment, I made some changes...